metafields_test.go

```diff
@@ -124,8 +124,60 @@ func TestMetafields_GetByKeyInvalid(t *testing.T) {
 	metafield.Key = ""
 	metafields := Metafields{metafield}
 
-	_, err := metafields.GetByKey("123")
+	_, err := metafields.GetByKey("123", "custom")
 	if err == nil {
 		t.Errorf("An error was thrown: %v", err)
 	}
 }
+
+/*
+GIVEN I have an array of metafields with one metafield that has "custom" namespace
+WHEN I ask to get metafields by this namespace
+THEN the length of the metafields array returned must be what is expected
+*/
+func TestMetafields_GetByNamespaceOneMetafield(t *testing.T) {
+	// Expects
+	fake := faker.New()
+	metafieldsNamespace := "custom"
+	expectedValue := 1
+
+	metafield := Metafield{}
+	fake.Struct().Fill(&metafield)
+	metafield.Namespace = metafieldsNamespace
+	metafields := Metafields{metafield}
+
+	returnedMetafields, err := metafields.GetByNamespace("custom")
+	if err != nil {
+		t.Errorf("An error was thrown: %v", err)
+	}
+
+	if !reflect.DeepEqual(len(returnedMetafields), expectedValue) {
```

Comment:
Is that actually checking the meta fields as it has `len()` on the first property? i would usually have a length check and the deep equal of what was returned as a second check.

Reply:
```golang
/*
GIVEN I have an array of metafields with one metafield that has "custom" namespace
WHEN I ask to get metafields by this namespace
THEN the length of the metafields array returned must be what is expected
*/
func TestMetafields_GetByNamespaceOneMetafield(t *testing.T) {
	// Expects
	fake := faker.New()
	metafieldsNamespace := "custom"
	expectedNoMetafields := 1

	metafield := Metafield{}
	fake.Struct().Fill(&metafield)
	metafield.Namespace = metafieldsNamespace
	metafields := Metafields{metafield}

	returnedMetafields, err := metafields.GetByNamespace("custom")
	if err != nil {
		t.Errorf("An error was thrown: %v", err)
	}

	if len(returnedMetafields) != expectedNoMetafields {
		t.Errorf("Was expecting: %v got %v", len(returnedMetafields), expectedNoMetafields)
	}

	if !reflect.DeepEqual(returnedMetafields, metafields) {
		t.Errorf("Was expecting: %v got %v", len(returnedMetafields), metafields)
	}
}
```